ThreeBotPackages/threefold/tfgrid_workloads/actors/workload_manager.py

```diff
@@ -22,20 +24,26 @@ class Meta:
             workload_id = pw.IntegerField(index=True, default=0)
             node_id = pw.IntegerField(index=True, default=0)
 
-        def trigger_func(model, obj, action, **kwargs):
-            if action == "save":
+        def index_create(model, obj, action, **kwargs):
+            if action == "set_post":
                 index = model.IndexTable.get_or_none(reservation_id=obj.id)
                 if not index:
-                    for workload_type in ["zdbs", "volumes", "containers"]:
-                        for workload in getattr(obj.data_reservation, workload_type):
-                            record = model.IndexTable.create(
-                                reservation_id=obj.id, workload_id=workload.workload_id, node_id=workload.node_id
-                            )
+                    workload_id = 1
+                    for _, workload in self._iterate_over_workloads(obj):
+                        index = model.IndexTable.create(
+                            reservation_id=obj.id, workload_id=workload_id, node_id=workload.node_id
```

Comment:
As far as I understood the spec. This is not up to us to full the workload_id from the reservation object.
The user (or the tool the users uses) must fill these when he creates the reservation. 
We should validate what we receive though. to make sure all the required field a populated and make sense.